File: routing-soft-ui/src/app/shared/interfaces/plan-insert-dto.ts
import { TruckReadOnlyDto } from "./truck-readonly-dto";
import { LocationNodeReadonlyDto } from "./location-node-readonly-dto";

export interface PlanInsertDto {
      trucksList: TruckReadOnlyDto[];
  locationNodeList: LocationNodeReadonlyDto[];
  algorithmSpec: string;
  timeslotLength: number;
}

File: routing-soft-core/src/main/java/org/routing/software/dtos/PlanInsertDto.java
package org.routing.software.dtos;


import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class PlanInsertDto {

    @NotEmpty(message = "The plan should have at least one truck")
    List<TruckReadOnlyDto> trucksList;

    @NotEmpty(message = "The plan should have at least one location node")
    List<LocationNodeReadOnlyDto> locationNodeList;

    String algorithmSpec;

    private Long timeslotLength;
}
